stdlib/2and3/filecmp.pyi

```diff
@@ -1,17 +1,31 @@
 # Stubs for filecmp (Python 2/3)
 import sys
+import os
 from typing import AnyStr, Callable, Dict, Generic, Iterable, List, Optional, Sequence, Tuple, Union, Text
 
 DEFAULT_IGNORES = ...  # type: List[str]
 
-def cmp(f1: Union[bytes, Text], f2: Union[bytes, Text], shallow: Union[int, bool] = ...) -> bool: ...
-def cmpfiles(a: AnyStr, b: AnyStr, common: Iterable[AnyStr],
-             shallow: Union[int, bool] = ...) -> Tuple[List[AnyStr], List[AnyStr], List[AnyStr]]: ...
+if sys.version_info >= (3, 6):
+    def cmp(f1: Union[bytes, Text, os.PathLike], f2: Union[bytes, Text, os.PathLike],
+            shallow: Union[int, bool] = ...) -> bool: ...
+    def cmpfiles(a: Union[AnyStr, os.PathLike], b: Union[AnyStr, os.PathLike], common: Iterable[Union[AnyStr, os.PathLike]],
```

Comment:
Because this uses AnyStr, we should ideally also use a type variable in the PathLike case (so `PathLike[AnyStr]`). But this sort of things tends to run into mypy bugs sometimes. We should check a few cases like these:
- `cmpfiles(bytes, bytes)` returns `Tuple[List[bytes], ...]`
- `cmpfiles(str, str)` returns `Tuple[List[str], ...]`
- `cmpfiles(PathLike[str], PathLike[str])` returns `Tuple[List[str], ...]`
- `cmpfiles(PathLike[str], bytes)` is an error

You can check these in mypy by using `reveal_type`.